### DISCUSSION TITLE
Can you change step logic for cypress plugin

### DISCUSSION THREAD

**Discussion by AndriiZhydkykh** [allure-framework/allure2#2481] [Questions & Support] [answered]
Can you change the logic of the steps? Now we should write the step function and put some cypress commands. It's take a lot of time
Can you change this function from this

it('displays two todo items by default', () => {
    step("baz1", () => {
      cy.get('.todo-list li').should('have.length', 2)
    });
    step("baz2", () => {
      cy.get('.todo-list li').first().should('have.text', 'Pay electric bill')
    });
    step("baz3", () => {
      cy.get('.todo-list li').last().should('have.text', 'Walk the dog')
    });

  })
  
  to this
  
  it('displays two todo items by default', () => {
    step("baz1" );
    cy.get('.todo-list li').should('have.length', 2)
    step("baz2")
    cy.get('.todo-list li').first().should('have.text', 'Pay electric bill')
    step("baz3")
    cy.get('.todo-list li').last().should('have.text', 'Walk the dogs')
  })

**Comment by baev** [comment, marked as answer]
The provided solution allows users to use nested steps, and it's done the same way in other frameworks, so it should be more convenient for users.

However, we have a feature concept called `stages` — a special type of step that ends when the next step/stage starts or the current executable (step/test) ends. Although, it is not yet implemented

**Reply by AndriiZhydkykh** [reply]
@baev Thanks. I will be waiting to realize the stage feature. Where I can see a change log to new realises?

**Reply by baev** [reply]
> I will be waiting to realize the stage feature.

It's not in our roadmap yet, but we'll discuss it internally and maybe increase the priority a bit. As for myself, I've been waiting for it for a long time already

> Where I can see a change log to new realises?

You can subscribe to the following repos:

https://github.com/allure-framework/allure-js — cypress integration
https://github.com/allure-framework/allure2 — allure report
